### PR TITLE
feat(wallet): Edit Visible Assets Network Selector

### PR DESCRIPTION
**Pull request by Douglashdaniel**
## Description 
- Adds the ability to filter the `Edit Visible Assets` by `network`
- Introduces `useTokenRegistry` hook we can start to migrate to and eventually remove `fullTokenList` from redux.

Resolves <https://github.com/brave/brave-browser/issues/23675>
Resolves <https://github.com/brave/brave-browser/issues/21770>

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

https://user-images.githubusercontent.com/40611140/196556853-83890347-f6de-40c0-bd90-b7037b4663f9.mov

### CODE REVIEW

**Review by onyb — COMMENTED** [review]
Is it useful to have an "All networks" option for Visible Assets modal? The default network in the dropdown should be the currently selected network.

**Comment by Douglashdaniel** [comment]
> Is it useful to have an "All networks" option for Visible Assets modal? The default network in the dropdown should be the currently selected network.

@onyb This was discussed/decided internally, we are working on getting away from having a global `Selected Network` in the wallet.

**Review by josheleonard — CHANGES_REQUESTED** [review]
We should also request this list directly from core to remove the need to store it in redux.

File: components/brave_wallet_ui/common/actions/wallet_actions.ts [hunk 1]
```diff
@@ -81,6 +82,7 @@ export const accountsChanged = createAction('accountsChanged')
 export const selectedAccountChanged = createAction<SelectedAccountChangedPayloadType>('selectedAccountChanged')
 export const setAllTokensList = createAction<BraveWallet.BlockchainToken[]>('setAllTokensList')
 export const getAllTokensList = createAction('getAllTokensList')
+export const setTokenRegistery = createAction<SetTokenRegistryType>('setTokenRegistery')
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/actions/wallet_actions.ts]: This has been Removed!

File: components/brave_wallet_ui/common/async/handlers.ts [hunk 2]
```diff
@@ -291,13 +291,15 @@ handler.on(WalletActions.getAllTokensList.getType(), async (store) => {
   const { blockchainRegistry } = getAPIProxy()
   const getAllTokensList = await Promise.all(networkList.map(async (network) => {
     const list = await blockchainRegistry.getAllTokens(network.chainId, network.coin)
-    return list.tokens.map((token) => {
+    const formatedList = list.tokens.map((token) => {
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/async/handlers.ts]: This has been removed!

**Review by josheleonard — CHANGES_REQUESTED** [review]

File: components/brave_wallet_ui/common/hooks/useTokenRegistry.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+// Types
+import { TokenRegistry, WalletState } from '../../constants/types'
+
+// Hooks
+import { useIsMounted, useLib } from './'
+
+export function useTokenRegistry () {
+  // Hooks
+  const isMounted = useIsMounted()
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/useTokenRegistry.tsx]: Done!

File: components/brave_wallet_ui/common/hooks/useTokenRegistry.tsx [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+              return {
+                ...token,
+                logo: `chrome://erc-token-images/${token.logo}`
+              }
+            })
+            registry[network.chainId] = formattedListWithIcons
+            if (!subscribed && !isMounted) {
+              return
+            }
+            setTokenRegistry(registry)
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/useTokenRegistry.tsx]: Done!

File: components/brave_wallet_ui/common/hooks/useTokenRegistry.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+export function useTokenRegistry () {
+  // Hooks
+  const isMounted = useIsMounted()
+  const { getTokenList } = useLib()
+
+  // Redux
+  const networkList = useSelector(({ wallet }: { wallet: WalletState }) => wallet.networkList)
+
+  // Hook State
```
[the remark this answers is not in the document]
Comment (Douglashdaniel) [reply components/brave_wallet_ui/common/hooks/useTokenRegistry.tsx:27]: Moved!

**Pull request merged**